pkg/store/prometheus.go

```diff
@@ -255,7 +255,7 @@ func (p *PrometheusStore) queryPrometheus(s storepb.Store_SeriesServer, r *store
 		}
 		matrix = result
 	} else {
-		vector, _, err := p.client.QueryInstant(s.Context(), p.base, r.ToPromQL(), timestamp.Time(r.MaxTime), opts)
+		vector, _, _, err := p.client.QueryInstant(s.Context(), p.base, r.ToPromQL(), timestamp.Time(r.MaxTime), opts)
```

Comment:
Hm, are we dropping a potential histogram response here? Is it possible to return a histogram?

Reply:
This is just the raw json message exposed as a workaround until https://github.com/prometheus/common/pull/417 is available, so we can check if response contains histogram: https://github.com/rabenhorst/thanos/blob/8edbf720be226a7df48eb49a10c03d792452dd53/pkg/promclient/promclient.go#L388